elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateStore.java

```diff
@@ -230,10 +242,97 @@ public void commit() {
                     compositeCriterion = Restrictions.and(compositeCriterion, criterion);
                 }
             }
+
             return compositeCriterion;
         }
 
         @Override
+        public Collection filterCollection(Collection collection, String type, FilterScope<?> filterScope) {
```

Comment:
Dan and I were originally thinking operators would be more like separate functions that take arguments (possibly variable number).  Each function would be responsible for building part of a query.  This would also make it easier to extend to add new operators.  This function has a large cyclomatic complexity.  I suggest we break it up - maybe in the manner of having an abstraction around operators.